### PR TITLE
support for visual editor reference links option

### PR DESCRIPTION
**Pull request by jjallaire**
@timtmok This is a change intended to support the visual editor's new `editor: markdown: references: links` option (as described here: https://quarto.org/docs/visual-editor/vscode/#links)

My Homebrew installation on my dev Mac has become corrupted so I'm not actually able to compile/test this! In any case we should pair program on landing the change so you know what's going on. Message me when a good time to do this next week might be (I'll mark this PR as a draft in the meantime).

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualModeMarkdownWriter.java [hunk 1]
```diff
@@ -121,6 +122,7 @@ public Options optionsFromConfig(PanmirrorPandocFormatConfig formatConfig)
                {
                   referencesLocationPref = quarto.project_editor.markdown.references.location;
                   referencesPrefixPref = quarto.project_editor.markdown.references.prefix;
+                  referencesLinksPref = quarto.project_editor.markdown.references.links;
```
Comment (timtmok) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualModeMarkdownWriter.java:125]: @jjallaire Everything looked good except this one part. `quarto.project_editor.markdown.references.links` is an array of length 1. If I set the project level value to `false`, the reference links are still created.

I'm not sure if this was fixed in a later Quarto release if it's parsing the yaml there. I'm using 1.2.335.

**Review by timtmok — COMMENTED** [review]
The document level option works. It was only the project level that didn't work when setting the option to `false`.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualModeMarkdownWriter.java (hunk 1 above)
Comment (jjallaire) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualModeMarkdownWriter.java:125]: Okay, I'll take a closer look at this (I now have a local RStudio dev config so can actually run!)

Comment (jjallaire) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualModeMarkdownWriter.java:125]: I just needed to add this bit to make sure `links` was read as a scalar: https://github.com/rstudio/rstudio/pull/12689/commits/d8a5aca44251cf2eb40893603b9b3d0f1fe282fe

**Comment by ronblum** [comment]
Note: I created issue #12478 to track this.

**Pull request merged**